cmd/awscollector/Dockerfile

```diff
@@ -4,42 +4,24 @@
 ################################
 FROM alpine:latest as build 
 
-ENV GOPROXY=direct
```

Comment:
After speaking with @Aneurysm9 we should not build the binary inside the docker container since make will build the binary and in ci we cache the binary.

Reply:
We were previously only building in the container when there was no prebuilt binary available, which isn't the case in the CI scripts. I think having a Dockerfile that can build without any external tooling / commands is quite useful, for example when developing I can just put a repo URL into a docker-compose file and things just work. A separate Dockerfile could work for this, but it seems like we can seamlessly support both cases with the previous approach.

Reply:
I would prefer @anuraaga 's approach since it would increase independence between files. However, if we planned used this Dockerfile only for the CI and make, then writing a notification with our customer about not supporting these would be a recommended way though.

Reply:
I'm good either way but I think I'll go with a separate dockerfile that builds the binary since I like how lean our current docker file is.

Reply:
Resolved